scrapi/base/helpers.py

```diff
@@ -57,8 +60,12 @@ def oai_extract_doi(identifiers):
 def oai_extract_url(identifiers):
     identifiers = [identifiers] if not isinstance(identifiers, list) else identifiers
     for item in identifiers:
-        if 'http://' in item or 'https://' in item and 'viewcontent' not in item:
-            return unicode(item)
+        try:
+            found_url = URL_REGEX.search(item).group()
+            if 'viewcontent' not in found_url:
+                return found_url.decode('utf-8')
+        except AttributeError:
```

Comment:
since this goes through every identifier tag looking for a url, .group() will raise an attribute error if it doesn't find a url in that particular identifier!